src/Item/Item.router.ts

```diff
@@ -199,126 +182,88 @@ export class ItemRouter extends Router {
       )
     }
 
-    const [dbItems, remoteItems] = await Promise.all([
+    const [
+      dbItems,
+      remoteItems,
+      { dbTPItems, remoteTPItems },
+    ] = await Promise.all([
       Item.find<ItemAttributes>({ eth_address }),
       collectionAPI.fetchItemsByAuthorizedUser(eth_address),
+      this.itemService.getTPItemsByManager(eth_address),
     ])
 
-    const catalystItems = await peerAPI.fetchWearables(
-      remoteItems.map((item) => item.urn)
-    )
+    const [items, tpItems] = await Promise.all([
+      Bridge.consolidateItems(dbItems, remoteItems),
+      Bridge.consolidateTPItems(dbTPItems, remoteTPItems),
+    ])
 
-    return Bridge.consolidateItems(dbItems, remoteItems, catalystItems)
+    // TODO: add sorting (we're not breaking pagination)
+    return items.concat(this.itemService.withOwner(tpItems, eth_address))
   }
 
-  async getItem(req: AuthRequest) {
+  getItem = async (req: AuthRequest): Promise<FullItem> => {
     const id = server.extractFromReq(req, 'id')
     const eth_address = req.auth.ethAddress
 
-    const dbItem = await Item.findOne<ItemAttributes>(id)
-    if (!dbItem) {
-      throw new HTTPError(
-        'Not found',
-        { id, eth_address },
-        STATUS_CODES.notFound
-      )
-    }
-
-    let fullItem: FullItem = Bridge.toFullItem(dbItem)
-    let fullCollection: CollectionAttributes | undefined = undefined
-
-    if (dbItem.collection_id && dbItem.blockchain_item_id) {
-      const dbCollection = await Collection.findOne<CollectionAttributes>(
-        dbItem.collection_id
-      )
+    try {
+      const { item, collection } = await this.itemService.getItem(id)
```

Comment:
Tried to follow the method firms I found on the other service methods like `deleteItem` and the like. I think I agree, as returning two things from a method is weird

Knowing this, you think it needs changing? If it can be done later, a comment?

Reply:
Definitely a comment would do it for now.